provider/cldr/src/transform/list/mod.rs

```diff
@@ -0,0 +1,169 @@
+    fn try_from(cldr_paths: &dyn CldrPaths) -> Result<Self, Self::Error> {
+        let mut data = vec![];
+        let path = cldr_paths.cldr_misc()?.join("main");
+        let locale_dirs = get_subdirectories(&path)?;
+        for dir in locale_dirs {
+            let path = dir.join("listPatterns.json");
+            let mut resource: cldr_serde::list_patterns_json::Resource =
+                serde_json::from_reader(open_reader(&path)?).map_err(|e| (e, path))?;
+            data.append(&mut resource.main.0);
+        }
+        Ok(Self { data })
+    }
+}
+
+impl KeyedDataProvider for ListProvider {
+    fn supports_key(resc_key: &ResourceKey) -> Result<(), DataError> {
+        key::LIST_FORMAT_V1.match_key(*resc_key)
+    }
+}
+
+impl<'data> DataProvider<'data, ListFormatterPatternsV1Marker> for ListProvider {
+    fn load_payload(
+        &self,
+        req: &DataRequest,
+    ) -> Result<DataResponse<'data, ListFormatterPatternsV1Marker>, DataError> {
+        Self::supports_key(&req.resource_path.key)?;
+        let cldr_langid: CldrLangID = req.try_langid()?.clone().into();
+        let patterns = match self
+            .data
+            .binary_search_by_key(&&cldr_langid, |(lid, _)| lid)
+        {
+            Ok(idx) => &self.data[idx].1.list_patterns,
+            Err(_) => return Err(DataError::MissingResourceOptions(req.clone())),
+        };
+
+        if cldr_langid.langid == langid!("es") {
+            unimplemented!("Add Spanish conditional data");
+        }
+
+        if cldr_langid.langid == langid!("iw") {
+            unimplemented!("Add Hebrew conditional data");
+        }
+
+        Ok(DataResponse {
+            metadata: DataResponseMetadata {
+                data_langid: req.resource_path.options.langid.clone(),
+            },
+            payload: Some(DataPayload::from_owned(
+                ListFormatterPatternsV1::try_from(patterns).unwrap(),
+            )),
+        })
+    }
+}
+
+icu_provider::impl_dyn_provider!(ListProvider, {
+    _ => ListFormatterPatternsV1Marker,
+}, SERDE_SE, 'data);
+
+impl<'data> IterableDataProviderCore for ListProvider {
+    fn supported_options_for_key(
+        &self,
+        _resc_key: &ResourceKey,
+    ) -> Result<Box<dyn Iterator<Item = ResourceOptions>>, DataError> {
+        let list: Vec<ResourceOptions> = self
+            .data
+            .iter()
+            .map(|(l, _)| ResourceOptions {
+                variant: None,
+                // TODO(#568): Avoid the clone
+                langid: Some(l.langid.clone()),
+            })
+            .collect();
+        Ok(Box::new(list.into_iter()))
+    }
+}
+
+impl<'a> TryFrom<&cldr_serde::list_patterns_json::ListPattern> for ListFormatterPattern<'a> {
+    type Error = icu_lists::error::Error;
+
+    fn try_from(other: &cldr_serde::list_patterns_json::ListPattern) -> Result<Self, Self::Error> {
+        Ok(Self {
+            start: ConditionalListJoinerPattern::from_str(&other.start)?,
```

Comment:
`nit`: `other.start.parse()?` here is a bit less mouthful :)

Reply:
And here I was wondering what the point of the FromStr trait is :)